ctapipe/io/metadata.py

```diff
@@ -0,0 +1,222 @@
+"""
+Management of CTA Reference Metadata, as defined in the CTA Top-Level Data Model
+document [ctatopleveldatamodel]_ , version 1A. This information is required to be
+attached to the header of any
+files generated.
+
+The class Reference collects all required reference metadata, and can be turned into a
+flat dictionary.  The user should try to fill out all fields, or use a helper to fill
+them (as in `Activity.from_provenance()`)
+
+.. code-block:: python
+
+    ref = Reference(
+        contact=Contact(name="Some User", email="user@me.com"),
+        product=Product(format='hdf5', ...),
+        process=Process(...),
+        activity=Activity(...),
+        instrument = Instrument(...)
+    )
+
+    some_astropy_table.meta = ref.to_dict()
+    some_astropy_table.write("output.ecsv")
+
+"""
+import uuid
+from collections import OrderedDict
+import warnings
+
+from astropy.time import Time
+from traitlets import (
+    Enum,
+    Unicode,
+    Int,
+    HasTraits,
+    default,
+    Instance,
+    validate,
+)
+
+from ctapipe.core.provenance import _ActivityProvenance
+
+__all__ = [
+    "Reference",
+    "Contact",
+    "Process",
+    "Product",
+    "Activity",
+    "Instrument",
+    "write_to_hdf5",
+]
+
+
+class Contact(HasTraits):
+    """ Contact information """
+
+    name = Unicode("unknown")
+    email = Unicode("unknown")
+    organization = Unicode("unknown")
+
+
+class Product(HasTraits):
+    """Data product information"""
+
+    description = Unicode("unknown")
+    creation_time = Unicode()
```

Comment:
Time traitlet?

Reply:
is there one? We could create one, but for this it may be overkill. This is just to be able to write header keywords to a file.  Would be nice to validate that it is in an astropy.Time format though, but that can come later.

Reply:
No, but we need one anyway probably at some point, right?

Reply:
done